src/archivist/recorder/index.js

```diff
@@ -0,0 +1,86 @@
+export default class Recorder {
+  constructor({ versionsStorageAdapter, snapshotsStorageAdapter }) {
+    if (!versionsStorageAdapter && !snapshotsStorageAdapter) {
+      throw new RangeError('storage adapters should be defined both for versions and snapshots');
```

Comment:
This is not what is currently checked.
```suggestion
    if (!versionsStorageAdapter || !snapshotsStorageAdapter) {
      throw new RangeError('Storage adapters should be defined both for versions and snapshots');
```